### PR TITLE
Oncoprint performance optimizations

### PR DESCRIPTION
**Pull request by adamabeshouse**
Decreases initial oncoprint rendering time of GENIE x 10 genes by half!

[Link](https://demo-quantum-portal-cbf747b6.cbioportal.org/results/oncoprint?cancer_study_list=genie_private_9_7&Z_SCORE_THRESHOLD=2.0&RPPA_SCORE_THRESHOLD=2.0&profileFilter=mutations%2Cfusion%2Ccna&case_set_id=genie_private_9_7_cnaseq&gene_list=SOX9%2520RAN%2520TNK2%2520EP300%2520PXN%2520NCOA2%2520AR%2520NRIP1%2520NCOR1%2520NCOR2&geneset_list=%20&tab_index=tab_visualize&Action=Submit)

Fixes https://github.com/cBioPortal/cbioportal/issues/8628

(1) Use webGL optimization to reduce cost of rendering all the default gray rectangles

(2) Reduce unnecessary repeat computation in initial oncoprint load

(3) Specify colors in numerical RGBA array, reducing surprisingly costly calls to parseInt and parseFloat in the oncoprintjs library

(4) Upgrade oncoprintjs version to include a lot of other optimizations which I did there, which don't need any corresponding adjustment in the frontend code

### CODE REVIEW

**Comment by alisman** [comment]
@adamabeshouse appears that structural variant data is lost here.  toggle between 3799 on Impact and you'll see
https://www.cbioportal.org/results/oncoprint?cancer_study_list=msk_impact_2017&Z_SCORE_THRESHOLD=2.0&RPPA_SCORE_THRESHOLD=2.0&profileFilter=mutations%2Cfusion%2Ccna&case_set_id=msk_impact_2017_cnaseq&gene_list=CDKN2A%2520MDM2%2520MDM4%2520TP53&geneset_list=%20&tab_index=tab_visualize&Action=Submit

**Comment by adamabeshouse** [comment]
Just found an issue - sometimes small oncoprints never get out of "rendering". maybe it is so fast now that its breaking something. Will look into it.

File: yarn.lock [hunk 1]
```diff
@@ -22615,13 +22612,6 @@ tinyqueue@^1.1.0:
   resolved "https://registry.yarnpkg.com/tinyqueue/-/tinyqueue-1.2.3.tgz#b6a61de23060584da29f82362e45df1ec7353f3d"
   integrity sha512-Qz9RgWuO9l8lT+Y9xvbzhPT2efIUIFd69N7eF7tJ9lnQl0iLj1M7peK7IoUGZL9DJHw9XftqLreccfxcQgYLxA==
```
Comment (alisman) [inline yarn.lock:22614]: @adamabeshouse this appears to be necessary

File: yarn.lock [hunk 2]
```diff
@@ -16994,7 +16992,6 @@ pathval@^1.1.0:
     react-toastify "^5.3.1"
     react-tooltip "^3.10.0"
     sharedb "^1.1.0"
-    tippy.js "^6.3.1"
```
Comment (alisman) [inline yarn.lock:16997]: does seem like it was removed here @adamabeshouse @onursumer.  @onursumer should it have been?

Comment (onursumer) [reply yarn.lock:16997]: @alisman @adamabeshouse This is still a dependency for pathway mapper, not sure why it was removed. There might be an issue with the latest pathway mapper version.

**Pull request merged**